cloudapp/src/app/item.service.ts

```diff
@@ -0,0 +1,42 @@
+import { Injectable } from '@angular/core';
+import { CloudAppEventsService, CloudAppRestService } from '@exlibris/exl-cloudapp-angular-lib';
+import { filter, map, mergeMap, switchMap } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ItemService {
+
+  constructor(
+    private restService: CloudAppRestService,
+    private eventsService: CloudAppEventsService,
+  ) { }
+
+  getItems() {
+    return this.eventsService.entities$.pipe(
+      filter(entities => entities.length > 0),
```

Comment:
@jatjhu Let me know if this makes sense. This is where a bulk of the logic is happening. 

Basically we want to get from the entities on the page to items which have all their bib and holdings metadata as well. This is so we can re-display them sorted with the RMST number which is at the item level.

Reply:
My main question is how the app would work. Are the users going to be using alma to scan and find the records, then we would use the events like this to perform operations on the results? Or is it going to be standalone without using alma to interact with the records?

Reply:
I think that the logic looks really good though. The subscribe to item service and pipe is pretty cool. Oh, I could see how the users would use alma and then have this running in the background and they could press a button or something to check it out or in.

Reply:
Wait is the item service just going to be looking for things that get added to the pick list in alma?

Reply:
This is pretty cool

Reply:
This is app is for printing pick lists. There's another cloud app we'll need to create for adding an RMST number to a tray of items.

The default one in Alma at Fulfillment > Pick from Shelf can't sort by the RMST number so this basically re-displays those items sorted in RMST order.